tidb-binlog/driver/reader/reader.go

```diff
@@ -152,14 +152,23 @@ func (r *Reader) run() {
 	}
 	defer partitionConsumer.Close()
 
+	// use variable msg to avoid message blocking while reading
+	var msg *Message
 	for {
+		var msgM chan *Message
+		var msgC <-chan *sarama.ConsumerMessage
+		if msg != nil {
+			msgM = r.msgs
```

Comment:
once reach this code, msgM is always r.msgs

Reply:
revised